batch/batch.go

```diff
@@ -0,0 +1,96 @@
+package batch
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/coreos/bbolt"
+)
+
+// errSolo is a sentinel error indicating that the requester should re-run the
+// operation in isolation.
+var errSolo = errors.New(
+	"batch function returned an error and should be re-run solo",
+)
+
+type request struct {
+	*Request
+	errChan chan error
+}
+
+type batch struct {
+	db     *bbolt.DB
+	start  sync.Once
+	reqs   []*request
+	clear  func(b *batch)
+	locker sync.Locker
+}
+
+// trigger is the entry point for the batch and ensures that run is started at
+// most once.
+func (b *batch) trigger() {
+	b.start.Do(b.run)
```

Comment:
Can it happen that `trigger` is called more than once? If not, does this `sync.Once` camouflage potential bugs?